mipengine/controller/api/algorithm_specifications_dtos.py

```diff
@@ -1,149 +1,78 @@
-    """
-    InputDataSpecificationsDTO is a superset of InputDataSpecifications
-    containing data_model, dataset and filter.
-    """
-
+class InputDataSpecificationsDTO(ImmutableBaseModel):
     data_model: InputDataSpecificationDTO
     datasets: InputDataSpecificationDTO
     filter: InputDataSpecificationDTO
     x: Optional[InputDataSpecificationDTO] = None
     y: Optional[InputDataSpecificationDTO] = None
 
-    def __init__(self, input_data_spec: InputDataSpecifications):
-        if input_data_spec.x:
-            self.x = InputDataSpecificationDTO(
-                label=input_data_spec.x.label,
-                desc=input_data_spec.x.desc,
-                types=input_data_spec.x.types,
-                notblank=input_data_spec.x.notblank,
-                multiple=input_data_spec.x.multiple,
-                stattypes=input_data_spec.x.stattypes,
-                enumslen=input_data_spec.x.enumslen,
-            )
-        if input_data_spec.y:
-            self.y = InputDataSpecificationDTO(
-                label=input_data_spec.y.label,
-                desc=input_data_spec.y.desc,
-                types=input_data_spec.y.types,
-                notblank=input_data_spec.y.notblank,
-                multiple=input_data_spec.y.multiple,
-                stattypes=input_data_spec.y.stattypes,
-                enumslen=input_data_spec.y.enumslen,
-            )
-        self.data_model = InputDataSpecificationDTO(
-            label="Data model of the data.",
-            desc="The data model that the algorithm will run on.",
-            types=["text"],
-            notblank=True,
-            multiple=False,
-            stattypes=None,
-            enumslen=None,
-        )
-        self.datasets = InputDataSpecificationDTO(
-            label="Set of data to use.",
-            desc="The set of data to run the algorithm on.",
-            types=["text"],
-            notblank=True,
-            multiple=True,
-            stattypes=None,
-            enumslen=None,
-        )
-        self.filter = InputDataSpecificationDTO(
-            label="filter on the data.",
-            desc="Features used in my algorithm.",
-            types=["jsonObject"],
-            notblank=False,
-            multiple=False,
-            stattypes=None,
-            enumslen=None,
-        )
-
-
-@dataclass_json
-@dataclass
-class ParameterSpecificationDTO(ParameterSpecification):
-    """
-    ParameterDTO is identical to the ParameterSpecification
-    but exists for consistency and future use if needed.
-    """
-
-    pass
-
-
-@dataclass_json
-@dataclass
-class AlgorithmSpecificationDTO:
-    """
-    AlgorithmDTO is used to provide the UI the requirements
-    of each algorithm.
-    System variables are added and unnecessary fields are removed.
-    """
 
+class ParameterSpecificationDTO(ImmutableBaseModel):
+    label: str
+    desc: str
+    type: ParameterType
+    notblank: bool
+    multiple: bool
+    default: Any
```

Comment:
Shouldn't the default be of type `ParameterType`?

Reply:
`default` is there to show a "default" value to the UI. It's not a `parameterType`, it could be "2" for example, or "lefthippocampus".